src/main/java/uk/gov/hmcts/reform/wataskmanagementapi/services/CFTTaskMapper.java

```diff
@@ -385,5 +363,35 @@ private WarningValues mapNoteResourceToWarnings(List<NoteResource> notes) {
 
         return value == null ? Optional.empty() : Optional.of((T) value);
     }
+
+
+    public Task mapToTask(TaskResource taskResource) {
+        return new Task(taskResource.getTaskId(),
+            taskResource.getTaskName(),
+            taskResource.getTaskType(),
+            taskResource.getState().getValue().toLowerCase(Locale.ROOT),
+            taskResource.getTaskSystem().getValue(),
+            taskResource.getSecurityClassification().getSecurityClassification(),
+            taskResource.getTitle(),
+            taskResource.getCreated().toZonedDateTime(),
+            taskResource.getDueDateTime().toZonedDateTime(),
+            taskResource.getAssignee(),
+            taskResource.getAutoAssigned(),
+            taskResource.getExecutionTypeCode().getExecutionName(),
+            taskResource.getJurisdiction(),
+            taskResource.getRegion(),
+            taskResource.getLocation(),
+            taskResource.getLocationName(),
+            taskResource.getCaseTypeId(),
+            taskResource.getCaseId(),
+            taskResource.getCaseCategory(),
+            taskResource.getCaseName(),
+            taskResource.getHasWarnings(),
+            mapNoteResourceToWarnings(taskResource.getNotes()),
+            taskResource.getCaseCategory(),
+            taskResource.getWorkTypeResource() == null ? null : taskResource.getWorkTypeResource().getId()
```

Comment:
the logic on line 140 extractWorkType() suggests that we always create a `new WorktypeResource()` under certain circumstances, it could even be `new WorktypeResource(null)` we need to review if this is the expected behaviour or if we want to throw an exception in that case however for this line since we are always instantiating an object, we should probably not do a ternary operation to default to null

Reply:
Once 2.1 is live there should always be a Work Type and we should fail fast if there's an initiation attempt without one.
However, historic Tasks won't have a work type in the DB so we have to handle that possibility and the front end should deal with that too.

Separately, could write a job which populates historic Tasks with a work type.

Reply:
Every new task has a work type and this code work for them. For historic tasks, they don't have a work type and in response we return WorkTypeResource like this **WorkTypeResource(id: null, label: "")** 
I think that, we don't need to change here, if not let me know tomorrow morning, please.

Reply:
```suggestion
            taskResource.getWorkTypeResource() == null ? null : taskResource.getWorkTypeResource()
```